ast/Helpers.h

```diff
@@ -355,10 +355,14 @@ class MK {
         return Send1(loc, T(loc), core::Names::unsafe(), std::move(inner));
     }
 
-    static TreePtr Untyped(core::LocOffsets loc) {
+    static TreePtr UntypedType(core::LocOffsets loc) {
         return Send0(loc, T(loc), core::Names::untyped());
     }
 
+    static TreePtr UntypedValue(core::LocOffsets loc) {
+        return Unsafe(loc, Nil(loc));
+    }
+
```

Comment:
My preference would have been to keep `MK::Untyped` named as is and to call this one `UnsafeNil`, which makes it more clear at the call site what's going on.

`T.unsafe(nil)` in DSL passes is generally a smell and we've been trying to move away from them, towards either rewriting it to what the runtime would actually do, or to a raise:

https://github.com/sorbet/sorbet/pull/2883/files#diff-0ebfafdc5884d2abe7cce00031996991cfb2bcdc67e666a7a34dab8052c4f2d2L113-R113

(Alternatively, if you wanted to try changing the places using your `MK::UntypedValue` helper to use `RaiseUnimplemented` that would be another way forwards.)